domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -2971,15 +3042,41 @@ public void getConnection(Handler<AsyncResult<PgConnection>> replyHandler) {
   }
 
   void getSQLConnection(Handler<AsyncResult<SQLConnection>> handler) {
+    getSQLConnection(0, handler);
+  }
+
+  void getSQLConnection(int queryTimeout, Handler<AsyncResult<SQLConnection>> handler) {
     getConnection(res -> {
       if (res.failed()) {
         handler.handle(Future.failedFuture(res.cause()));
         return;
       }
-      handler.handle(Future.succeededFuture(new SQLConnection(res.result(), null)));
+      SQLConnection sqlConnection = new SQLConnection(res.result(), null);
+      if (queryTimeout > 0) {
+        long timerId = vertx.setTimer(queryTimeout, id -> sqlConnection.conn.cancelRequest(ar -> {
+          if (ar.succeeded()) {
+            log.warn(
+                String.format("Cancelling request due to timeout after : %d ms",
+                    queryTimeout));
+          } else {
+            log.warn("Failed to send cancelling request", ar.cause());
+          }
+        }));
+        activeConnectionsTimers.put(sqlConnection, timerId);
+      }
+      handler.handle(Future.succeededFuture(sqlConnection));
     });
   }
 
+  private void cancelConnectionTimeoutTimer(SQLConnection sqlConnection) {
+    Long timerId = activeConnectionsTimers.remove(sqlConnection);
+    if (timerId != null) {
+      vertx.cancelTimer(timerId);
+    } else {
+      log.warn("Cannot cancel timer: " + timerId);
```

Comment:
There should be no warning if cancelling was disabled with queryTimeout==0.

Reply:
done